Marlin/src/feature/fwretract.cpp

```diff
@@ -121,16 +123,10 @@ void FWRetract::retract(const bool retracting
   //*/
 
   const bool has_zhop = retract_zlift > 0.01;     // Is there a hop set?
-
   const float old_feedrate_mm_s = feedrate_mm_s;
-  const int16_t old_flow = planner.flow_percentage[active_extruder];
-
-  // Don't apply flow multiplication to retract/recover
-  planner.flow_percentage[active_extruder] = 100;
```

Comment:
Our big mistake , to add this , the flow must remains exactly what it is
LCD never updated in real time , because of too much retractation flow modification
LCD issues to set a flow value while retractions are activated , because stored and restored , and your lcd value is lost

Reply:
Retraction / recover should honor the mm length given. So what we have to do instead is to increase the E motion in proportion to the current flow so that it will extrude the specified length.

So…

```cpp
current_position[E_AXIS] +=
  ((swapping ? swap_retract_length : retract_length) / planner.volumetric_multiplier[active_extruder])
  / (planner.flow_percentage[active_extruder] * 0.01);
```